hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -455,44 +459,47 @@ public void removeTrigger(String tenantId, String triggerId) {
     }
 
     private void removeTrigger(Trigger trigger) {
+        final String tenantId = trigger.getTenantId();
+        final String triggerId = trigger.getId();
+
+        // If necessary, clean up working memory
```

Comment:
I see that alertsEngineCache and missingStates are update out of this edge.
Probably is safe, but I wonder motivation, is there any potential case where a trigger to remove from the alerts engine is not found on the drools engine ?.
I ask for this, because if you found a case that this is true, probably this is something related to other problem.